src/lib/types/restrictions.ts

```diff
@@ -0,0 +1,112 @@
+import { z } from 'zod'
```

Comment:
why typescript was not efficient enough?

Reply:
because `createProcess` takes user input JSON we need a runtime validation library to check the JSON contains valid `Restrictions`. `zod` errors are quite descriptive too
```
     ZodError: [
  {
    "code": "invalid_type",
    "expected": "number",
    "received": "undefined",
    "path": [
      "FixedInputMetadataValue",
      0,
      "index"
    ],
    "message": "Required"
  },
]
  ```